core/lib/dal/migrations/20250213120658_create_etherscan_verification_requests_table.up.sql

```diff
@@ -0,0 +1,20 @@
+CREATE TABLE IF NOT EXISTS etherscan_verification_requests (
+    contract_verification_request_id BIGINT PRIMARY KEY,
+    status CHARACTER VARYING(32) NOT NULL,
+    processing_started_at TIMESTAMP,
+    attempts INT NOT NULL DEFAULT 0,
+    etherscan_verification_id CHARACTER VARYING(128),
+    result TEXT,
+    error TEXT,
+    created_at TIMESTAMP NOT NULL,
+    updated_at TIMESTAMP NOT NULL,
+    CONSTRAINT contract_verification_requests_id_fkey FOREIGN KEY (contract_verification_request_id) REFERENCES public.contract_verification_requests (id) ON UPDATE NO ACTION ON DELETE NO ACTION
```

Comment:
Just to understand better, how will `ON DELETE NO ACTION` work? Does it mean that if there is a row in `etherscan_verification_requests` with id=5 and there is a query `DELETE FROM contract_verification_requests WHERE id = 5` it just fails?

Reply:
Yes, that's how it works with ON DELETE NO ACTION. You are unable to delete a record from contract_verification_requests if there is a record in etherscan_verification_requests that references it (via a foreign key).

If you need to delete a record, there are two ways:

1. Delete the record from etherscan_verification_requests first, then delete the record from contract_verification_requests.
2. Delete both records within the same DB transaction. It doesn't even matter which one is removed first, as the foreign key constraint is checked at transaction commit and won't be violated since both records are deleted.